mavconn/mavconn.py

```diff
@@ -26,5 +44,59 @@ def clear_handler(self, message_name=None):
         else:
             self._stacks.clear()
 
-    def add_timer(period, handler):
-        pass
+    def add_timer(self, period, handler):
+        heappush(self._timers, Timer(period, handler))
+
+    def timer_work(self, cv):
+        with cv:
+            cv.wait_for(self._timers != [])
+            current_timer = heappop(self._timers)
+            current_timer.wait_time()
+            current_timer.handle()
+            heappush(self._timers, current_timer)
+
+class Timer:
+    """Definition of Timer class"""
+
+    def __init__(self, period, handler):
+        self._period = period
+        self._handler = handler
+        current_time = datetime.datetime.now()
+        period_seconds = timedelta(seconds=self._period)
+        self._next_time = current_time + period_seconds
+
+    def wait_time(self):
```

Comment:
Just copy this into the beginning of handle.  Or call it there, you should not have to manually call it timer_work.